NightscoutServiceKit/NightscoutService.swift

```diff
@@ -59,432 +49,96 @@ public final class NightscoutService: Service {
-        guard let uploader = uploader else {
-            completion(entries)
-            return
-        }
+    public var pumpEventDataLimit: Int? { return 1000 }
 
-        uploader.deleteCarbEntries(entries, completion: completion)
+    public func uploadPumpEventData(_ stored: [PersistedPumpEvent], completion: @escaping (Result<Bool, Error>) -> Void) {
+        completion(.success(false))
     }
 
-}
+    public var settingsDataLimit: Int? { return 1000 }
 
-private extension Array where Element == RepeatingScheduleValue<Double> {
-    func scheduleItems() -> [ProfileSet.ScheduleItem] {
-        return map { (item) -> ProfileSet.ScheduleItem in
-            return ProfileSet.ScheduleItem(offset: item.startTime, value: item.value)
-        }
+    public func uploadSettingsData(_ stored: [StoredSettings], completion: @escaping (Result<Bool, Error>) -> Void) {
+        activeUploader!.uploadProfiles(stored.compactMap { ProfileSet(storedSettings: $0) }, completion: completion)
     }
-}
 
-private extension LoopKit.TemporaryScheduleOverride {
-    func nsScheduleOverride(for unit: HKUnit) -> NightscoutUploadKit.TemporaryScheduleOverride {
-        let nsTargetRange: ClosedRange<Double>?
-        if let targetRange = settings.targetRange {
-            nsTargetRange = ClosedRange(uncheckedBounds: (
-                lower: targetRange.lowerBound.doubleValue(for: unit),
-                upper: targetRange.upperBound.doubleValue(for: unit)))
-        } else {
-            nsTargetRange = nil
-        }
-
-        let nsDuration: TimeInterval
-        switch duration {
-        case .finite(let interval):
-            nsDuration = interval
-        case .indefinite:
-            nsDuration = 0
-        }
+    public var statusDataLimit: Int? { return 1000 }
 
-        return NightscoutUploadKit.TemporaryScheduleOverride(
-            targetRange: nsTargetRange,
-            insulinNeedsScaleFactor: settings.insulinNeedsScaleFactor,
-            symbol: context.symbol,
-            duration: nsDuration,
-            name: context.name)
+    public func uploadStatusData(_ stored: [StoredStatus], completion: @escaping (Result<Bool, Error>) -> Void) {
+        activeUploader!.uploadDeviceStatuses(stored.map { DeviceStatus(storedStatus: $0) }, completion: completion)
     }
-}
 
-private extension LoopKit.TemporaryScheduleOverride.Context {
-    var name: String? {
-        switch self {
-        case .custom:
-            return nil
-        case .legacyWorkout:
-            return LocalizedString("Workout", comment: "Name uploaded to Nightscout for legacy workout override")
-        case .preMeal:
-            return LocalizedString("Pre-Meal", comment: "Name uploaded to Nightscout for Pre-Meal override")
-        case .preset(let preset):
-            return preset.name
+    private var activeUploader: NightscoutUploader? {
```

Comment:
nit: Make `uploader` a lazy var?

Reply:
It is a computed property (because it is in an extension) so, by definition, it is lazy. I can move it to the class definition and make it a lazy stored property if you wish.

Let me know.

Reply:
I was just suggesting making uploader an actual lazy var, so there aren't two different properties that readers have to try to understand what's different between them, when to use one vs the other, etc.

Reply:
Ah, got it! Yea, that's definitely better.

Reply:
Done.